cameo/flux_analysis/analysis.py

```diff
@@ -367,29 +378,139 @@ class _PhenotypicPhasePlaneChunkEvaluator(object):
     def __init__(self, model, variable_reactions):
         self.model = model
         self.variable_reactions = variable_reactions
+        objective_reactions = [reaction for reaction in model.reactions
+                               if reaction.objective_coefficient != 0]
+        if len(objective_reactions) != 1:
+            raise Exception('multiple objectives not supported')
+        self.product_reaction = objective_reactions[0]
+        self.c_source_reactions = self.get_c_source_reactions()
+
+    def get_c_source_reactions(self):
```

Comment:
The objective is not changed during FVA/phenotypic_phase_plane. It is always what `objective=` specifies. Also, if you have multiple carbon sources you cannot compute mass or molar yields, only c-mol/c-mol yields. In your example it would simply be `product_flux/(A_flux * A.n_carbons + F_flux * F.n_carbons)`.